packages/actor-extract-links-solid-type-index/lib/ActorExtractLinksSolidTypeIndex.ts

```diff
@@ -144,6 +143,53 @@ export class ActorExtractLinksSolidTypeIndex extends ActorExtractLinks {
     return typeLinks;
   }
 
+  /**
+   * To fetch the domain of the predicate.
+   * @param predicateSubjectRec A dictionary of predicate and its subjects from the query.
+   * @param typeSubjects A dictionary of class type and its subjects from the query.
+   */
+  public async linkPredicateDomain(predicateSubjectRec: Record<string, RDF.Term>,
+    typeSubjects: Record<string, RDF.Term[]>): Promise<void> {
+    const predicateDomains = await this.getRDFTypeFromPredicates(Object.keys(predicateSubjectRec));
+
+    for (const [ predicate, subject ] of Object.entries(predicateSubjectRec)) {
+      // If multiple subjects have the same predicate domain.
+      const typeName = predicateDomains[predicate];
+      if (typeName) {
+        if (!typeSubjects[typeName]) {
+          typeSubjects[typeName] = [];
+        }
+        typeSubjects[typeName].push(subject);
+      }
+    }
+  }
+
+  /**
+   * To fetch the rdf type from the vocabulary if the type is not already present.
+   * @param predicateValue Array of predicate values from the query.
+   * @return predicateTypeLinks A record mapping predicate URIs to it's domain.
+   */
+  public async getRDFTypeFromPredicates(predicateValue: string[]): Promise<Record<string, string>> {
+    const vocabSource: [IDataSource, ...IDataSource[]] =
+      [ predicateValue[0], ...predicateValue.slice(1).toString().split(',') ];
```

Comment:
```suggestion
      <[IDataSource, ...IDataSource[]]> predicateValue;
```
With perhaps just an if-check before to ensure that length > 0.